DataDog/datadog-operator
# Load environment variables in override with `envFrom`

### What does this PR do?

Add ability for environment variables to be loaded in the override map of datadog-agent.yaml using `envFrom`. 
Example:
<img width="331" alt="Screenshot 2024-09-23 at 4 52 29 PM" src="https://github.com/user-attachments/assets/f09c6c1f-772c-4915-9787-b413aec34408">

### Motivation

#896 

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* make a kind cluster with Operator
* Make and load some test configMaps and secrets into kind cluster. To double check the priority between env and envFrom (env should take priority), add a field `DD_CLUSTER_NAME: <not your cluster name>`
* Add fields into `datadog-agent.yaml`: spec.override.nodeAgent.EnvFrom, spec.override.clusterAgent.envFrom, and/or spec.override.clusterChecksRunner.envFrom with configMaps or secrets
* add Agent to kind cluster
* run `kubectl describe pod` for agent, cluster checks runner, and cluster agent and verify that the configs or secrets that you specified are there. They should be in a separate section right above the environment variables. 
* run `kubectl exec -it <datadog agent pod> bash` and then `env`. You should be able to see the specific variables you added with their values.
* For agent that you modified DD_CLUSTER_NAME for, when you run the exec command, the `DD_CLUSTER_NAME` field should not be overridden to what was put in the configMap/secret.

### Checklist
